conslogging/conslogging.go

```diff
@@ -239,12 +239,24 @@ func (cl ConsoleLogger) prettyPrefix() string {
 	}
 
 	formatString := fmt.Sprintf("%%%vv", cl.prefixPadding)
-	substringStart := len(cl.prefix) - cl.prefixPadding
 
-	clampedPadding := substringStart
-	if clampedPadding < 0 {
-		clampedPadding = 0
+	prettyPrefix := cl.prefix
+	if len(cl.prefix) > cl.prefixPadding {
+		parts := strings.Split(cl.prefix, "/")
+		target := parts[len(parts)-1]
+
+		truncated := ""
+		for _, part := range parts[:len(parts)-1] {
+			letter := ""
+			if len(part) > 0 {
+				letter = string(part[0])
+			}
```

Comment:
I just realized - we should avoid truncating `..` to `.`

Reply:
Hm interesting; I didnt realize we could have relative paths? What if we ran them through https://golang.org/pkg/path/#Clean before truncation, this wouldn't be an issue then?

Reply:
This is for the local target case, not remote. For example this `FROM` here https://github.com/earthly/earthly/blob/master/examples/tests/Earthfile#L1. It wouldn't work for that unfortunately.